src/main/java/nextstep/subway/section/domain/Sections.java

```diff
@@ -39,4 +51,108 @@ public int size() {
+                    .orElseThrow(() -> new IllegalArgumentException(NOT_FOUND_SORTED_NEXT_STATION));
+            sortedAllStations.add(nextStation);
+        }
+        return LineStations.from(sortedAllStations);
+    }
+
+    private boolean isEdgeSection(Section addedSection) {
+        return isEdgeUpSection(addedSection) || isEdgeDownSection(addedSection);
+    }
+
+    private boolean isEdgeUpSection(Section addedSection) {
+        return findEdgeUpSection().getUpStation().equals(addedSection.getDownStation());
+    }
+
+    private boolean isEdgeDownSection(Section addedSection) {
+        return findEdgeDownSection().getDownStation().equals(addedSection.getUpStation());
+    }
+
+    private Section findEdgeUpSection() {
+        List<Station> downStations = this.sections.stream()
+                .map(Section::getDownStation)
+                .collect(Collectors.toList());
+        return this.sections.stream()
+                .filter(section -> !section.isIncludedUpStation(downStations))
+                .findFirst()
+                .orElseThrow(() -> new IllegalArgumentException(NOT_FOUND_EDGE_UP_SECTION));
+    }
+
+    private Section findEdgeDownSection() {
+        List<Station> upStations = this.sections.stream()
+                .map(Section::getUpStation)
+                .collect(Collectors.toList());
+        return this.sections.stream()
+                .filter(section -> !section.isIncludedDownStation(upStations))
+                .findFirst()
+                .orElseThrow(() -> new IllegalArgumentException(NOT_FOUND_EDGE_DOWN_SECTION));
+    }
+
+    private void relocateConnectSection(Section addedSection) {
+        if (!this.sections.isEmpty() && !isEdgeSection(addedSection)) {
+            Section connectSection = findConnectSection(addedSection);
+            relocateUpSection(connectSection, addedSection);
+            relocateDownSection(connectSection, addedSection);
+        }
+    }
+
+    private void relocateUpSection(Section connectSection, Section addedSection) {
+        if (connectSection.isEqualsDownStation(addedSection.getDownStation())) {
+            connectSection.updateDownStation(addedSection.getUpStation());
+            connectSection.decreaseDistance(addedSection.getDistance());
+        }
+    }
+
+    private void relocateDownSection(Section connectSection, Section addedSection) {
+        if (connectSection.isEqualsUpStation(addedSection.getUpStation())) {
+            connectSection.updateUpStation(addedSection.getDownStation());
+            connectSection.decreaseDistance(addedSection.getDistance());
+        }
+    }
+
+    private Section findConnectSection(Section section) {
+        return findSectionByUpStation(section.getUpStation())
+                .orElseGet(() -> findSectionByDownStation(section.getDownStation())
+                    .orElseThrow(() -> new IllegalArgumentException(NOT_CONTAINS_CONNECT_SECTION))
+                );
```

Comment:
stream 구문 depth가 일정하지 않은 부분을 말씀드린 내용이었습니다 ☺️
```suggestion
        return findSectionByUpStation(section.getUpStation()).orElseGet(() -> findSectionByDownStation(section.getDownStation())
                .orElseThrow(() -> new IllegalArgumentException(NOT_CONTAINS_CONNECT_SECTION)));
```